helm-quarry/templates/redis_deployment.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: apps/v1
```

Comment:
You are correct that a redis restart would cause users to be logged out. I'm not sure if it would cause celery worker issues or not. If it would quarry should be updated to something that shouldn't experience the same (https://phabricator.wikimedia.org/T278583 remains open as this remains possible, the stop button helped, but doesn't truly resolve the issue.)

From a deployment standpoint it is a lot more stuff to put out if we leave redis on a VM. If indeed the only issue is logouts, I see that as a reasonable exchange. Should a hosted solution happen, we can switch to that. Though perhaps we would have discovered it was not necessary.

Further we will be able to test everything in a full k8s deploy before any of the VM deploy goes down. So if it is found that a redis restart is a significant problem, we can focus on a VM option at that time.

I propose we leave this here until it proves to be a clear issue.

Reply:
Okay.

If we do see any issues, we could look into enabling AOF persistence. That should mean no data loss, and the db size is small enough so that it should also not unduly slow down startup. The downside would be the AOF file needing a PersistentVolume which in our case would be the NFS, which could be kinda slow - but not sure if that would be a dealbreaker as appends are asynchronous and non-blocking.